core/src/autogluon/core/hpo/executors.py

```diff
@@ -0,0 +1,370 @@
+        model_name
+            The model name of the hpo experiment is tuning. This is only used for logging purpose
+        """
+        raise NotImplementedError
+    
+    @abstractmethod
+    def execute(self, **kwargs):
+        """Execute the experiment"""
+        raise NotImplementedError
+    
+    @abstractmethod
+    def report(
+            self,
+            reporter: 'LocalReporter',
+            **kwargs
+        ):
+        """Report result of the experiment to the reporter. If no reporter needed, pass in None"""
+        raise NotImplementedError
+    
+    @abstractmethod
+    def get_hpo_results(
+            self,
+            model_name: str,
+            model_path_root: str,
+            **kwargs
+        ):
+        """
+        Retrieve hpo results
+        
+        Parameters
+        ----------
+        model_name
+            The model name of the hpo experiment is tuning.
+        model_path_root
+            Root path of the model
+        """
+        raise NotImplementedError
+
+
+class RayHpoExecutor(HpoExecutor):
+    """Implementation of HpoExecutor Interface, where ray tune is used as the backend"""
+    
+    custom_to_ray_preset_map = {
+        'auto': {'scheduler': 'FIFO', 'searcher': 'bayes'},
+        'local_random': {'scheduler': 'FIFO', 'searcher': 'random'},
+        'random': {'scheduler': 'FIFO', 'searcher': 'random'},
+    }
+    custom_to_ray_scheduler_preset_map = {
+        'local': 'FIFO',
+    }
+    custom_to_ray_searcher_preset_map = {
+        'local_random': 'random',
+        'random': 'random',
+        'auto': 'random',
+    }
+    
+    def __init__(self):
+        self.resources = None
+        self.hyperparameter_tune_kwargs = None
+        self.analysis = None
+    
+    @property
+    def executor_type(self):
+        return RAY_BACKEND
+    
+    def initialize(self, hyperparameter_tune_kwargs, default_num_trials=None, time_limit=None):
+        self.time_limit = time_limit
+        if isinstance(hyperparameter_tune_kwargs, dict):
+            hyperparameter_tune_kwargs = hyperparameter_tune_kwargs.copy()
+        if isinstance(hyperparameter_tune_kwargs, str):
+            hyperparameter_tune_kwargs = self.custom_to_ray_preset_map[hyperparameter_tune_kwargs]
+        hyperparameter_tune_kwargs['scheduler'] = self.custom_to_ray_scheduler_preset_map.get(
+            hyperparameter_tune_kwargs['scheduler'],
+            hyperparameter_tune_kwargs['scheduler']
+        )
+        hyperparameter_tune_kwargs['searcher'] = self.custom_to_ray_searcher_preset_map.get(
+            hyperparameter_tune_kwargs['searcher'],
+            hyperparameter_tune_kwargs['searcher']
+        )
+        if 'num_trials' not in hyperparameter_tune_kwargs and default_num_trials is not None:
+            hyperparameter_tune_kwargs['num_trials'] = default_num_trials
+        self.hyperparameter_tune_kwargs = hyperparameter_tune_kwargs
+        return time_limit
```

Comment:
Strange return statement. Why return time_limit?

Reply:
Actually I think it's not needed